src/pages/lessons/[slug].tsx

```diff
@@ -420,157 +407,166 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
-                  console.debug(`player ready [autoplay:${autoplay}]`)
-                  const player: HTMLVideoElement =
-                    event.target as HTMLVideoElement
-                  player.volume = volumeRate / 100
-                  player.playbackRate = playbackRate
-                  actualPlayerRef.current = player
-                  const isDifferent = lastAutoPlayed.current !== lesson?.hls_url
-                  if (autoplay && isDifferent && isFunction(player.play)) {
-                    console.debug(`autoplaying`)
-                    lastAutoPlayed.current = lesson?.hls_url
-                    player.play()
-                  }
-                }}
-                onPause={() => {
-                  send('PAUSE')
-                }}
-                onPlay={() => send('PLAY')}
-                onTimeUpdate={(event: any) => {
-                  onProgress(
-                    {playedSeconds: event.target.currentTime},
-                    lesson,
-                  ).then((lessonView: any) => {
-                    if (lessonView) {
-                      console.debug('progress recorded', {
-                        progress: lessonView,
-                      })
-                      setLessonView(lessonView)
-                    }
-                  })
-                }}
-                onEnded={() => {
-                  console.debug(`received ended event from player`)
-                  send('COMPLETE')
-                }}
-                onVolumeChange={(event: SyntheticEvent) => {
-                  const player: HTMLVideoElement =
-                    event.target as HTMLVideoElement
-                  setPlayerPrefs({volumeRate: player.volume * 100})
-                }}
-              />
-            </PlayerContainer>
-            {spinnerVisible && (
-              <div className="flex justify-center items-center absolute z-10 top-0 right-0 bottom-0 left-0 bg-black bg-opacity-80">
-                <Spinner />
-              </div>
-            )}
-
-            {playerState.matches('joining') && (
-              <OverlayWrapper>
-                <CreateAccountCTA
-                  lesson={get(lesson, 'slug')}
-                  technology={primary_tag}
-                />
-              </OverlayWrapper>
-            )}
-            {playerState.matches('subscribing') && (
-              <OverlayWrapper>
-                <JoinCTA lesson={lesson} />
-              </OverlayWrapper>
-            )}
-            {playerState.matches('pitchingCourse') && (
-              <OverlayWrapper>
-                <CoursePitchOverlay
-                  lesson={lesson}
-                  onClickRewatch={() => {
-                    send('VIEW')
-                    if (actualPlayerRef.current) {
-                      actualPlayerRef.current.play()
-                    }
+            <div
+              className={`relative ${
+                isFullscreen ? 'lg:col-span-12' : 'lg:col-span-9'
```

Comment:
we had local state for fullscreen so I adapted this to use that instead of needing player

having `usePlayer` above caused a lot of redrawing!